src/cljs/main/broadfcui/page/workspace/notebooks/notebooks_table.cljs

```diff
@@ -0,0 +1,512 @@
+   [broadfcui.utils.user :as user]
+   [broadfcui.common.components :as comps]
+   [broadfcui.common.icons :as icons]
+   [broadfcui.common.flex-utils :as flex]
+   [broadfcui.common.links :as links]
+   [broadfcui.common.style :as style]
+   [broadfcui.common.table :refer [Table]]
+   [broadfcui.common.table.style :as table-style]
+   [broadfcui.common.table.utils :as table-utils]
+   [broadfcui.config :as config]
+   [broadfcui.components.blocker :refer [blocker]]
+   [broadfcui.components.buttons :as buttons]
+   [broadfcui.components.foundation-dropdown :as dropdown]
+   [broadfcui.components.modals :as modals]
+   [broadfcui.components.spinner :refer [spinner]]
+   [broadfcui.endpoints :as endpoints]
+   [broadfcui.page.workspace.notebooks.create_notebook :refer [NotebookCreator]]
+   [broadfcui.page.workspace.notebooks.create_cluster :refer [ClusterCreator]]
+   [broadfcui.page.workspace.notebooks.delete_cluster :refer [ClusterDeleter]]
+   [broadfcui.page.workspace.notebooks.cluster_details :refer [ClusterDetailsViewer]]
+   [broadfcui.page.workspace.notebooks.choose_cluster :refer [ChooseClusterViewer]]
+   [broadfcui.page.workspace.notebooks.cluster_error :refer [ClusterErrorViewer]]
+   [broadfcui.page.workspace.notebooks.rename_notebook :refer [NotebookRenamer]]
+   [broadfcui.page.workspace.notebooks.delete_notebook :refer [NotebookDeleter]]
+   [broadfcui.page.workspace.notebooks.duplicate_notebook :refer [NotebookDuplicator]]
+   [broadfcui.page.workspace.monitor.common :as moncommon]
+   [broadfcui.page.workspace.notebooks.utils :as notebook-utils]
+   ))
+
+(def spinner-icon
+  [:span {:style {:display "inline-flex" :alignItems "center" :justifyContent "center" :verticalAlign "middle"
+                  :width table-style/table-icon-size :height table-style/table-icon-size
+                  :borderRadius 3 :margin "-4px 4px 0 0"}
+          :data-test-id "status-icon" :data-test-value "unknown"}
+   (spinner)])
+
+(def pause-icon
+  [:span {:style {:display "inline-flex" :alignItems "center" :justifyContent "center" :verticalAlign "middle"
+                  :width table-style/table-icon-size :height table-style/table-icon-size
+                  :borderRadius 3 :margin "-4px 4px 0 0"}
+          :data-test-id "status-icon" :data-test-value "unknown"}
+   (icons/render-icon {} :pause)])
+
+(def play-icon
+  [:span {:style {:display "inline-flex" :alignItems "center" :justifyContent "center" :verticalAlign "middle"
+                  :width table-style/table-icon-size :height table-style/table-icon-size
+                  :borderRadius 3 :margin "-4px 4px 0 0"}
+          :data-test-id "status-icon" :data-test-value "unknown"}
+   (icons/render-icon {} :play)])
+
+(react/defc NotebooksTable
+  {:render
+   (fn [{:keys [state props this]}]
+     (let [{:keys [choose-notebook]} @state
+           {:keys [notebooks cluster-map pet-token toolbar-items]} props]
+       [:div {}
+        (when (:show-cluster-choose-dialog? @state)
```

Comment:
There's a utility `broadfcui.components.modals/show-modals` for handling large numbers of modals like this. It also sticks the `:dismiss` on there for you.